migrations/20230210165254_assessment_instances__include_in_statistics__update.sql

```diff
@@ -0,0 +1,44 @@
+WITH
+assessment_instances_with_users AS (
+    -- Get all assessment instances with all users who own them, either directly
+    -- or via group membership. Note that a single assessment instance can
+    -- appear multiple times, once for each group user.
+    SELECT
+        ai.id AS assessment_instance_id,
+        coalesce(ai.user_id, gu.user_id) AS user_id
+    FROM
+        assessment_instances AS ai
+        JOIN assessments AS a ON (a.id = ai.assessment_id)
+        JOIN course_instances AS ci ON (ci.id = a.course_instance_id)
+        LEFT JOIN groups AS g ON (g.id = ai.group_id AND g.deleted_at IS NULL)
+        LEFT JOIN group_users AS gu ON (gu.group_id = g.id)
+),
+instructor_assessment_instances AS (
+    -- For each of the previous assessment instances with users, select only the
+    -- ones with users who are instructors in the course, copying the logic from
+    -- sprocs/users_is_instructor_in_course_instance.sql. Then only keep
+    -- distinct assessment instances.
```

Comment:
Checking my understanding: this means that if a group consists of 5 students and an instructor, that assessment will be marked as `include_in_statistics = FALSE`, correct?

Reply:
That's correct. This is not necessarily the same as what the new code will set for `include_in_statistics` because the new code using the instructor-status of whichever user creates the assessment instance to decide. I think in practice that this shouldn't make much difference either way.